mscore/propertymenu.cpp

```diff
@@ -588,5 +558,84 @@ void ScoreView::editTremoloBarProperties(TremoloBar* tb)
+                              score()->undoChangeClef(part->staff(i), element, clefType, ic);
+                              }
+                        }
+                  // Change key signature if necessary
+                  if (it->transpose != oldV) {
+                        for (int i = 0; i < part->nstaves(); i++) {
+                              if (!part->staff(i)->keySigEvent(tickStart).isAtonal()) {
+                                    KeySigEvent ks;
+                                    ks.setForInstrumentChange(true);
+                                    Key key = part->staff(i)->key(tickStart);
+                                    if (!score()->styleB(Sid::concertPitch))
+                                          key = transposeKey(key, oldV);
+                                    ks.setKey(key);
+                                    score()->undoChangeKeySig(part->staff(i), tickStart, ks, ic);
+                                    }
+                              }
+                        }
+                  // change instrument in all linked scores
+                  for (ScoreElement* se : ic->linkList()) {
+                        InstrumentChange* lic = static_cast<InstrumentChange*>(se);
+                        Instrument* instrument = new Instrument(Instrument::fromTemplate(it));
+                        lic->score()->undo(new ChangeInstrument(lic, instrument));
+                        }
+                  
+                  // transpose for current score only
+                  // this automatically propagates to linked scores
+                  if (part->instrument(tickStart)->transpose() != oldV) {
+                        auto i = part->instruments()->upper_bound(tickStart.ticks());    // find(), ++i
+                        Fraction tickEnd;
+                        if (i == part->instruments()->end())
+                              tickEnd = Fraction(-1, 1);
+                        else
+                              tickEnd = Fraction::fromTicks(i->first);
+                        ic->score()->transpositionChanged(part, oldV, tickStart, tickEnd);
+                        }
+			
+                  if (ic->warning()) {
+                        ic->warning()->setPlainText(it->trackName);
+                        }
+                  else if (InstrumentChange* prevIc = score()->prevInstrumentChange(ic->segment()->prev1(), ic->staff(), true)) {
+                        StaffText* staffText = prevIc->warning();
+                        prevIc->setWarning(nullptr);
+                        staffText->setPlainText(it->trackName);
+                        ic->setWarning(staffText);
+                        }
+                  else {
+                        Chord* nextChord = ic->score()->nextChord(ic->segment(), ic->staff());
+                        if (nextChord) {
+                              ic->setNextChord(nextChord);
+                              }
```

Comment:
This however looks like the correct logic to me. Why are we even checking for a previous instrument change in the first place, in the above part?